File: gaia-sdk-javascript/src/graphql/response/type/Statement.ts


import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../../request/enumeration/RuntimeState";
import {SkillState} from "../../request/enumeration/SkillState";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Represents statement information
*/
export interface Statement {
    /**
    * The statement id
    */
    identityId?:Uuid, 
    /**
    * The statement reference id
    */
    reference?:Uuid, 
    /**
    * The name of the statement
    */
    qualifier?:string, 
    /**
    * Detailed description about the statement
    */
    appendent?:string, 
    /**
    * The utterance dictionary. The key is a language key and the value is a list of utterances
    */
    utterance?:Struct, 
    /**
    * The list of labels of the statement
    */
    labelList?:[string], 
    /**
    * The version of the statement
    */
    version?:string
}

File: gaia-sdk-python/graphql/response/type/Statement.py



class Statement:
    """
    this type represents the statement information
    """
    """
    The statement id
    """
    def identityId(self) -> Uuid:
        return self.identityId
    """
    The statement reference id
    """
    def reference(self) -> Uuid:
        return self.reference
    """
    The name of the statement
    """
    def qualifier(self) -> String:
        return self.qualifier
    """
    Detailed description about the statement
    """
    def appendent(self) -> String:
        return self.appendent
    """
    The utterance dictionary. The key is a language key and the value is a list of utterances
    """
    def utterance(self) -> Struct:
        return self.utterance
    """
    The list of labels of the statement
    """
    def labellist(self) -> String:
        return self.labellist
    """
    The version of the statement
    """
    def version(self) -> String:
        return self.version
